client/state/analytics/actions.js

```diff
@@ -0,0 +1,47 @@
+import fromPairs from 'lodash/fromPairs';
+
+import {
+	BUMP_STAT,
+	RECORD_EVENT,
+	RECORD_PAGE_VIEW
+} from './constants.js';
+
+export const withAnalytics = ( analytics, action ) =>
```

Comment:
> Also, as a sidenote:
> Please, keep in mind that actionCreators can be thunks in which case this will fail miserably :)

Very true! Unfortunately I can't imagine any way in which it would be wise to try and follow the thunk.

Maybe I should throw an exception if this gets a function. I'm a bit worried about this in general though because that's a runtime condition and may not be easily seen in development.

Maybe with our powers combined we can remove `redux-thunk`!